quantum/via.c

```diff
@@ -0,0 +1,382 @@
+        {
+            dynamic_keymap_set_keycode( command_data[0], command_data[1], command_data[2], ( command_data[3] << 8 ) | command_data[4] );
+            break;
+        }
+        case id_dynamic_keymap_reset:
+        {
+            dynamic_keymap_reset();
+            break;
+        }
+        case id_backlight_config_set_value:
+        case id_backlight_config_get_value:
+        case id_backlight_config_save:
+        {
+            raw_hid_receive_kb(data, length);
+            break;
+        }
+        case id_dynamic_keymap_macro_get_count:
+        {
+            command_data[0] = dynamic_keymap_macro_get_count();
+            break;
+        }
+        case id_dynamic_keymap_macro_get_buffer_size:
+        {
+            uint16_t size = dynamic_keymap_macro_get_buffer_size();
+            command_data[0] = size >> 8;
+            command_data[1] = size & 0xFF;
+            break;
+        }
+        case id_dynamic_keymap_macro_get_buffer:
+        {
+            uint16_t offset = ( command_data[0] << 8 ) | command_data[1];
+            uint16_t size = command_data[2]; // size <= 28
+            dynamic_keymap_macro_get_buffer( offset, size, &command_data[3] );
+            break;
+        }
+        case id_dynamic_keymap_macro_set_buffer:
+        {
+            uint16_t offset = ( command_data[0] << 8 ) | command_data[1];
+            uint16_t size = command_data[2]; // size <= 28
+            dynamic_keymap_macro_set_buffer( offset, size, &command_data[3] );
+            break;
+        }
+        case id_dynamic_keymap_macro_reset:
+        {
+            dynamic_keymap_macro_reset();
+            break;
+        }
+        case id_dynamic_keymap_get_layer_count:
+        {
+            command_data[0] = dynamic_keymap_get_layer_count();
+            break;
+        }
+        case id_dynamic_keymap_get_buffer:
+        {
+            uint16_t offset = ( command_data[0] << 8 ) | command_data[1];
+            uint16_t size = command_data[2]; // size <= 28
+            dynamic_keymap_get_buffer( offset, size, &command_data[3] );
+            break;
+        }
+        case id_dynamic_keymap_set_buffer:
+        {
+            uint16_t offset = ( command_data[0] << 8 ) | command_data[1];
+            uint16_t size = command_data[2]; // size <= 28
+            dynamic_keymap_set_buffer( offset, size, &command_data[3] );
+            break;
+        }
+        case id_eeprom_reset:
+        {
+            via_eeprom_reset();
+            break;
+        }
+        case id_bootloader_jump:
+        {
+            // Need to send data back before the jump
+            // Informs host that the command is handled
+            raw_hid_send( data, length );
+            // Give host time to read it
+            wait_ms(100);
+            bootloader_jump();
+            break;
+        }
+        default:
+        {
+            // The command ID is not known
+            // Return the unhandled state
```

Comment:
Is this something we also want to fall through to be handled by KB or is that something we explicitly don’t want to happen?